hermes-consumers/src/test/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSenderTest.java

```diff
@@ -330,10 +331,28 @@ public void shouldDeliverToModifiedEndpoint() {
         verify(otherMessageSender).send(message);
     }
 
+    @Test
+    public void shouldDeliverToNewSenderAfterModifiedTimeout() {
+        // given
+        Message message = message();
+        Subscription subscriptionWithModfiedTimeout = subscriptionWithRequestTimeout(2000);
+        MessageSender otherMessageSender = mock(MessageSender.class);
+
+        when(messageSenderFactory.create(subscriptionWithModfiedTimeout)).thenReturn(otherMessageSender);
+        when(otherMessageSender.send(message)).thenReturn(success());
+
+        // when
+        sender.updateSubscription(subscriptionWithModfiedTimeout);
+        sender.sendMessage(message);
+
+        // then
+        verify(otherMessageSender).send(message);
+    }
+
```

Comment:
I added test for timeout to JettyMessageSenderTest... The whole thing is very funky to test though...